radis/lbl/base.py

```diff
@@ -2421,88 +2446,127 @@ def calc_populations_noneq(
+    
+                        # reindexing to get a direct access to Qrot database
+                        # create the lookup dictionary
+                        # dfQrot index is already 'viblvl'
+                        dfQrot_dict = dict(list(zip(dfQrot.index, dfQrot.Qrot)))
+    
+                        dg = df.loc[idx]
+    
+                        # Add lower state Qrot
+                        dg_sorted = dg.set_index(["viblvl_l"], inplace=False)
+                        df.loc[idx, "Qrotl"] = dg_sorted.index.map(dfQrot_dict.get).values
+                        # Add upper state energy
+                        dg_sorted = dg.set_index(["viblvl_u"], inplace=False)
+                        df.loc[idx, "Qrotu"] = dg_sorted.index.map(dfQrot_dict.get).values
+    
+                        # ... note: the .map() version is about 3x faster than
+                        # ... dg.groupby('viblvl_l').apply(lambda x: dfQrot_dict[x.name])
+    
+                        if radis.DEBUG_MODE:
+                            assert (df.loc[idx, "iso"] == iso).all()
 
         self.profiler.stop("part_function", "partition functions")
         self.profiler.start("population", 3)
 
         # %%
 
         #  Derive populations
-        # ... vibrational distribution
-        if vib_distribution == "boltzmann":
-            # equation generated with @pytexit.py2tex > see docstrings.
-            df["nu_vib"] = df.gvibu / df.Qvib * exp(-hc_k * df.Evibu / Tvib)
-            df["nl_vib"] = df.gvibl / df.Qvib * exp(-hc_k * df.Evibl / Tvib)
-        elif vib_distribution == "treanor":
-            df["nu_vib"] = (
-                df.gvibu
-                / df.Qvib
-                * exp(-hc_k * (df.Evibu_h / Tvib + df.Evibu_a / Trot))
-            )
-            df["nl_vib"] = (
-                df.gvibl
-                / df.Qvib
-                * exp(-hc_k * (df.Evibl_h / Tvib + df.Evibl_a / Trot))
-            )
-        else:
-            raise ValueError(
-                "Unknown vibrational distribution: {0}".format(vib_distribution)
-            )
-
-        # ... Add vibrational-specific overpopulation factors
-        if overpopulation != {}:
-            for viblvl, ov in overpopulation.items():
-                if ov != 1:
-                    df.loc[df.viblvl_u == viblvl, "nu_vib"] *= ov
-                    df.loc[df.viblvl_l == viblvl, "nl_vib"] *= ov
+        if self.misc.parsum_mode == 'tabulation':
+            # ... vibrational distribution
+            if vib_distribution == "boltzmann":
+                df["nu_vib_x_Qvib"] = df.gvibu * exp(-hc_k * df.Evibu / Tvib)
+                df["nl_vib_x_Qvib"] = df.gvibl * exp(-hc_k * df.Evibl / Tvib)
+            elif vib_distribution == "treanor":
+                raise NotImplementedError("TO DO!") #!!!TODO 
+            else:
+                raise ValueError(
+                    "Unknown vibrational distribution: {0}".format(vib_distribution)
+                )
+                
```

Comment:
I think you're missing
```
if overpopulation != {}:
```
here. Which should be; at least; "NotImplementedError()", or even the actuel overpopulations (just copy-paste the block from below)